game_store module

metabobb opened the pull request:
**Changes to game_market:**
- game_market ui/ux improvements
- added server opcode support 238
- added client opcode support 233
**Changes to compiling on linux:**
- fixes compile errors on linux (kernel: 4.15.0-47-generic). Tested on Ubuntu 18.04
**New game module:**
- game_store module

metabobb commented:
Note Travis build fails with `g++: error: unrecognized command line option ‘-no-pie’`. Could be due to compiler version mismatch. Travis uses `gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4` while I tested on `gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0`.

hrsantiago commented:
@metabobb Most of your code seems fine, however there are major code style issues.
Also, there's a lot of different stuff on a single PR.

metabobb commented:
Review the code if you want and I'll make the necessary changes. You can also create a pull request on my repo to fix whatever issues there might be.

Also, sorry that it's so much stuff. With regards to `game_market` I have added a minor UI/UX change and made the client to update market data automatically wherever possible. Other commits seem self-explanatory to me.

I've made a thread on OTLand explaining how to test: https://otland.net/threads/release-game_store-module.264142/

eduardodantas95 commented:
Thanks @metabobb it solve my problem, this: https://otland.net/threads/otc-market-bug-and-mounts-bug.264163/#post-2552817

Thank you very much ... I tried to fix this mistake for days, and apparently I would never get it, since I was looking in a totally wrong direction. Its modifications in the market and protocol have totally solved.
Now the market opens without any problems. I will test the other things and give my report, for now I have only to thank and I hope that more people like you appear in the Tibian community.

metabobb commented:
@eduardodantas95 Yes, I have fixed that and also added support for two missing opcodes for client & server. I have tested the market with these changes and it run normally without any issues.

eduardodantas95 commented:
> @eduardodantas95 Yes, I have fixed that and also added support for two missing opcodes for client & server. I have tested the market with these changes and it run normally without any issues.

I tested everyone, everything working perfectly as you reported. Thank you very much for contributing.
